### DISCUSSION TITLE
Raising integration events

### DISCUSSION THREAD

**Discussion by PilotBob** [jasontaylordev/CleanArchitecture#661] [Q&A]
Hi. Are there any examples of raising integration events in the template? 

ie, domain events are fired in the savechanges and become part of that unit of work.

I would like to fire events AFTER the save changes worked. For example, to pass an event/message to an external system. Or to send a notification/email. Things that should only happen if the save is successful.

Unless I'm missing this, I don't see anything in the template that allows for this.

**Comment by deceptiveSimplicity** [comment]
You could save an integration event to some kind of an outbox, see outbox pattern. This way the integration event would be sent after SaveChanges.

**Reply by PilotBob** [reply]
Yes, I have a few ideas on how to do it. Trying to decide if I re-raise the same events as integration events, or create an integration event from the domain events, or wrap the domain event with an integration event. 

I was just wondering if maybe it would be good to add some guidance and examples to the template. (probably too much to bring in some type of messages buss)
